Patch by Robert Scheck <robert@fedoraproject.org> for ucview >= 0.33, which
adds missing linking to gmodule-2.0 during build. Without this patch ucview
rebuild on Fedora 17+ simply fails with:

  ucview-ucview-window.o: undefined reference to symbol 'g_module_name'

The second part of this patch is run-time only, however it avoids running
autoconf in the buildsystem.

--- ucview-0.33/configure.in		2010-10-04 19:46:32.000000000 +0200
+++ ucview-0.33/configure.in.gmodule	2012-01-15 19:27:03.000000000 +0100
@@ -37,7 +37,7 @@
 
 
 PKG_CHECK_MODULES(UCVIEW,
-		  libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 )
+		  libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 )
 
 #CPPFLAGS="${CPPFLAGS} -DG_DISABLE_DEPRECATED -DGDK_PIXBUF_DISABLE_DEPRECATED -DGDK_DISABLE_DEPRECATED -DGTK_DISABLE_DEPRECATED"
 
--- ucview-0.33/configure		2010-10-04 19:46:47.000000000 +0200
+++ ucview-0.33/configure.gmodule	2012-01-15 19:26:46.000000000 +0100
@@ -11564,12 +11564,12 @@
         pkg_cv_UCVIEW_CFLAGS="$UCVIEW_CFLAGS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 \""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 \""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_UCVIEW_CFLAGS=`$PKG_CONFIG --cflags "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 " 2>/dev/null`
+  pkg_cv_UCVIEW_CFLAGS=`$PKG_CONFIG --cflags "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 " 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -11582,12 +11582,12 @@
         pkg_cv_UCVIEW_LIBS="$UCVIEW_LIBS"
     else
         if test -n "$PKG_CONFIG" && \
-    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 \""; } >&5
-  ($PKG_CONFIG --exists --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ") 2>&5
+    { { $as_echo "$as_me:${as_lineno-$LINENO}: \$PKG_CONFIG --exists --print-errors \"libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 \""; } >&5
+  ($PKG_CONFIG --exists --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ") 2>&5
   ac_status=$?
   $as_echo "$as_me:${as_lineno-$LINENO}: \$? = $ac_status" >&5
   test $ac_status = 0; }; then
-  pkg_cv_UCVIEW_LIBS=`$PKG_CONFIG --libs "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 " 2>/dev/null`
+  pkg_cv_UCVIEW_LIBS=`$PKG_CONFIG --libs "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 " 2>/dev/null`
 else
   pkg_failed=yes
 fi
@@ -11606,14 +11606,14 @@
         _pkg_short_errors_supported=no
 fi
         if test $_pkg_short_errors_supported = yes; then
-	        UCVIEW_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 "`
+	        UCVIEW_PKG_ERRORS=`$PKG_CONFIG --short-errors --errors-to-stdout --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 "`
         else
-	        UCVIEW_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 "`
+	        UCVIEW_PKG_ERRORS=`$PKG_CONFIG --errors-to-stdout --print-errors "libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 "`
         fi
 	# Put the nasty error message in config.log where it belongs
 	echo "$UCVIEW_PKG_ERRORS" >&5
 
-	as_fn_error "Package requirements (libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ) were not met:
+	as_fn_error "Package requirements (libunicapgtk >= 0.2.23 gtk+-2.0 >= 2.8.0 gmodule-2.0 >= 2.8.0 gthread-2.0 >= 2.8.0 gconf-2.0 libucil >= 0.2.2 libglade-2.0 ) were not met:
 
 $UCVIEW_PKG_ERRORS
 
